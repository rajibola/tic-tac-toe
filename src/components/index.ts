export * from './board';
export * from './square';
export * from './round_button';
