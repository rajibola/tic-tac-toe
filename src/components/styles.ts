import {Platform, StyleSheet} from 'react-native';
import {colors} from '../constants/colors';
import {font} from '../constants/font';
import {getStatusBarHeight, hp, wp} from '../utils/layout';

export const RoundButttonStyles = StyleSheet.create({
  title: {
    ...font[1],
    textTransform: 'uppercase',
    fontWeight: 'bold',
    color: colors.white,
  },
  container: {
    height: hp(40),
    paddingHorizontal: wp(10),
    width: wp(120),
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.blue,
    borderRadius: wp(2),
  },
});

export const GameStyles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    alignSelf: 'center',
    marginTop: hp(25),
    width: wp(300),
  },
  vs: {
    ...font[1],
    lineHeight: hp(55),
    fontSize: hp(50),
    marginHorizontal: wp(4),
    letterSpacing: -wp(5),
  },
  textContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  container: {
    flex: 1,
    paddingTop: Platform.OS === 'ios' ? getStatusBarHeight() : 0,
  },
  player: {
    alignSelf: 'center',
    fontSize: hp(35),
    fontFamily: 'YanoneKaffeesatz-Regular',
    textTransform: 'lowercase',
  },
});

export const SquareStyles = StyleSheet.create({
  text: {
    fontSize: hp(50),
  },
  container: {
    borderWidth: wp(2),
    width: wp(100),
    height: wp(100),
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: -wp(2),
    marginTop: -wp(2),
    borderColor: colors.dark,
  },
});

export const BoardStyles = StyleSheet.create({
  boxContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: wp(300),
    alignSelf: 'center',
  },
});
