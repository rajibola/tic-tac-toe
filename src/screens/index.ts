export * from './game';
export * from './home';
export * from './intro';
