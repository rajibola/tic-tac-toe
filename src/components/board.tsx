import React from 'react';
import {Text, View} from 'react-native';
import {Square} from '.';
import {BoardStyles as styles} from './styles';

export const Board: React.FC = props => {
  const status = 'Next player: X';
  return (
    <View>
      <Text>{status}</Text>
      <View style={styles.boxContainer}>
        {Array(9)
          .fill(null)
          .map((a, i) => (
            <Square box={i} />
          ))}
      </View>
    </View>
  );
};
