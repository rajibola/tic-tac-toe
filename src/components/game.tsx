import React from 'react';
import {Text, View} from 'react-native';
import {Board} from '.';
import {GameStyles as styles} from './styles';

export const Game: React.FC = props => {
  return (
    <View style={styles.container}>
      <Text>Game</Text>
      <Board />
    </View>
  );
};
