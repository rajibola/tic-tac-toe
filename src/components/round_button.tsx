import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {RoundButttonProps} from '../types/types.d';
import {RoundButttonStyles as styles} from './styles';

export const RoundButtton: React.FC<RoundButttonProps> = ({
  style,
  onPress,
  title,
  titleStyles,
}) => {
  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      <Text style={[styles.title, titleStyles]}>{title}</Text>
    </TouchableOpacity>
  );
};
