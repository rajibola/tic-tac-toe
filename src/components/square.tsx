import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {SquareStyles as styles} from './styles';

type Square = {
  box: number | string;
  onPress: () => void;
};

export const Square: React.FC<Square> = props => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress}
      activeOpacity={0.8}>
      <Text style={styles.text}>{props.box}</Text>
    </TouchableOpacity>
  );
};
