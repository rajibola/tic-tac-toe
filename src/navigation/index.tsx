import * as React from 'react';
import 'react-native-gesture-handler';
import {StackNavigationOptions} from '@react-navigation/stack';
import {
  createSharedElementStackNavigator,
  SharedElementsComponentConfig,
} from 'react-navigation-shared-element';
import {NavigationParamList} from '../types/types.d';
import {Game, Home, Intro} from '../screens';

const options: StackNavigationOptions = {
  headerShown: false,
  cardStyleInterpolator: ({current: {progress}}) => {
    return {
      cardStyle: {
        opacity: progress,
      },
    };
  },
  gestureEnabled: false,
};

const SharedConfig: SharedElementsComponentConfig = route => {
  const {names} = route.params;
  console.log(names);
  return [
    {
      id: `item.${names[0]}.playerOne`,
      animation: 'move',
    },
    {
      id: `item.${names[1]}.playerTwo`,
      animation: 'move',
    },
  ];
};

const RootStack = createSharedElementStackNavigator<NavigationParamList>();

export default function MainNavigator() {
  return (
    <RootStack.Navigator initialRouteName="Home" screenOptions={options}>
      <RootStack.Screen
        name="Game"
        component={Game}
        sharedElementsConfig={SharedConfig}
      />
      <RootStack.Screen name="Home" component={Home} />
      <RootStack.Screen name="Intro" component={Intro} />
    </RootStack.Navigator>
  );
}
