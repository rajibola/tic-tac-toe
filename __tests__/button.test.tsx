import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';
import {Text} from 'react-native';
import {RoundButtton} from '../src/components';

describe('Button', () => {
  it('Button component renders correctly', () => {
    const text = 'Login';
    const instance = renderer.create(
      <RoundButtton title={text} onPress={() => null} />,
    );
    expect(instance).toMatchSnapshot();
  });
  it('RoundButtton component renders correctly after an update', () => {
    const value1 = 'Hello';
    const value2 = 'Hi';

    const instance = renderer.create(
      <RoundButtton title={value1} onPress={() => null} />,
    );
    const textInstance = instance.root.findByType(Text);
    expect(textInstance.props.children).toBe(value1);

    instance.update(<RoundButtton title={value2} onPress={() => null} />);
    expect(textInstance.props.children).toBe(value2);
  });

  it('RoundButtton responds to onPress', () => {
    const mockFn = jest.fn();

    const {getByText} = render(<RoundButtton title="TEST" onPress={mockFn} />);

    fireEvent.press(getByText('TEST'));

    expect(mockFn).toBeCalled();
  });
});
