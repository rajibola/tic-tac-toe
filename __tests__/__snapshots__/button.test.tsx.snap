// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#192a56",
      "borderRadius": 3.5,
      "borderWidth": 1,
      "height": 59.5,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 18,
      "width": 217.5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "YanoneKaffeesatz-Regular",
        "fontSize": 37,
        "fontWeight": "bold",
        "lineHeight": 59.5,
        "textTransform": "uppercase",
      }
    }
  >
    Login
  </Text>
</View>
`;
